### PR TITLE
Fix macOS app bundle name in packaging scripts

### LINKED ISSUE #3679: Case mismatch in path to Nextcloud.app in MacOS package script for client >= 3.3.0
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Description

In #3404 the macOS installation path changed from `/Applications/nextcloud.app` (up to 3.2.4) to `/Applications/Nextcloud.app` (from 3.3.0 onwards). However, the change in capitalization was not reflected in packaging scripts.

Use the following commands to extract `preinstall` and `postinstall` scripts from released packages:

```
export VERSION=3.3.0
mkdir /tmp/nextcloud-$VERSION
/tmp/nextcloud-$VERSION.pkg
curl -LO "https://github.com/nextcloud/desktop/releases/download/v$VERSION/Nextcloud-$VERSION.pkg"
xar -xf Nextcloud-$VERSION
cpio -zi < Nextcloud.pkg/Scripts
cat postinstall
cat preinstall
```

The resulting `preinstall` and `postinstall` scripts all refer the pre `3.3.0` path `/Applications/nextcloud.app` and also the old binary name `nextcloud` at various places.

### Client configuration
Client version: 3.3.0, 3.3.1

Operating system: macOS (any version)

OS language: any

Installation path of client: /Applications/Nextcloud.app

### PR DESCRIPTION
**Pull request by znerol**
Closes #3679
Follow-up for #3404

### CODE REVIEW

**Comment by FlexW** [comment]
Thank you for seeing that!

**Review by FlexW — APPROVED** [review]

File: admin/osx/pre_install.sh.cmake [hunk 1]
```diff
@@ -2,5 +2,6 @@
 
 # kill the old version. see issue #2044
 killall @APPLICATION_EXECUTABLE@
```
Comment (mgallien) [inline admin/osx/pre_install.sh.cmake:4]: why do we need to keep this line ?

Comment (znerol) [reply admin/osx/pre_install.sh.cmake:4]: The `preinstall` script opportunistically tries to kill a possibly running client. However, it cannot tell whether the previous binary is named `nextcloud` (pre 3.3.0) or `Nextcloud`. In fact, it doesn`t care if there was a client installed in the first place.

In order to cater for all cases, this PR changes the `preinstall` script to just try to kill any of `nextcloud` and `Nextcloud`.

Comment (mgallien) [reply admin/osx/pre_install.sh.cmake:4]: OK

**Review by mgallien — APPROVED** [review]

**Comment by FlexW** [comment]
\rebase

**Comment by FlexW** [comment]
@znerol Could rebase? Then we can merge.

**Comment by znerol** [comment]
Rebase done.

**Comment by FlexW** [comment]
\backport to stable-3.3

**Pull request merged**